traits/tests/test_trait_list_object.py

```diff
@@ -0,0 +1,350 @@
+
+    def test_setitem(self):
+        tl = TraitList([1, 2, 3],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl[1] = 5
+        self.assertEqual(1, self.index)
+        self.assertEqual([2], self.removed)
+        self.assertEqual([5], self.added)
+
+        tl[:] = [1, 2, 3, 4, 5]
+        self.assertEqual(slice(0, 3, None), self.index)
+        self.assertEqual([1, 5, 3], self.removed)
+        self.assertEqual([1, 2, 3, 4, 5], self.added)
+
+    def test_delitem(self):
+        tl = TraitList([1, 2, 3],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        del tl[2]
+        self.assertEqual(2, self.index)
+        self.assertEqual([3], self.removed)
+        self.assertEqual([], self.added)
+
+        del tl[:]
+        self.assertEqual(slice(0, 2, None), self.index)
+        self.assertEqual([1, 2], self.removed)
+        self.assertEqual([], self.added)
+
+        with self.assertRaises(IndexError):
+            del tl[0]
+
+    def test_iadd(self):
+        tl = TraitList([4, 5],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl += [6, 7]
+        self.assertEqual(slice(2, 4, None), self.index)
+        self.assertEqual([], self.removed)
+        self.assertEqual([6, 7], self.added)
+
+    def test_imul(self):
+        tl = TraitList([1, 2],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl *= 1
+        self.assertListEqual([1, 2], tl)
+        self.assertEqual(None, self.index)
+        self.assertEqual(None, self.removed)
+        self.assertEqual(None, self.added)
+
+        tl *= 2
+        self.assertEqual(slice(2, 4, None), self.index)
+        self.assertEqual([], self.removed)
+        self.assertEqual([1, 2], self.added)
+
+        with self.assertRaises(TypeError):
+            tl *= "5"
+
+        with self.assertRaises(TypeError):
+            tl *= 2.5
+
+        tl *= -1
+        self.assertEqual(slice(0, 4, None), self.index)
+        self.assertEqual([1, 2, 1, 2], self.removed)
+        self.assertEqual([], self.added)
+
+    def test_append(self):
+        tl = TraitList([1],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl.append(2)
+        self.assertEqual(1, self.index)
+        self.assertEqual([], self.removed)
+        self.assertEqual([2], self.added)
+
+    def test_extend(self):
+        tl = TraitList([1],
+                       validator=int_validator,
+                       notifiers=[self.notification_handler])
+
+        tl.extend([1, 2])
+        # self.assertEqual(slice(1, 3, None), self.index)
+        self.assertEqual([], self.removed)
+        self.assertEqual([1, 2], self.added)
```

Comment:
I'd really prefer to have all these `assertEqual` calls be consistent with (most of) the rest of the codebase (and with the majority of other Python projects): `actual, expected` rather than `expected, actual`. At some level it doesn't matter which we choose, but it's jarring for the reader to have a mixture of both.